src/shared_realm.hpp

```diff
@@ -384,6 +384,11 @@ class InvalidEncryptionKeyException : public std::logic_error {
 public:
     InvalidEncryptionKeyException() : std::logic_error("Encryption key must be 64 bytes.") {}
 };
+
+class InvalidDuplicateHandoverException : public std::logic_error {
```

Comment:
I don't think we need a distinct exception class for this unless we expect it to require special handling in the bindings. Using `std::logic_error` directly is fine.